packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/DateTimeWidget.tsx

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { getCSSColor, useSelection } from '@eclipse-sirius/sirius-components-core';
+import { DateTimeStyleProps } from '@eclipse-sirius/sirius-components-forms';
+import TextField from '@material-ui/core/TextField';
+import Typography from '@material-ui/core/Typography';
+import { Theme, makeStyles } from '@material-ui/core/styles';
+import HelpOutlineOutlined from '@material-ui/icons/HelpOutlineOutlined';
+import { useEffect, useRef, useState } from 'react';
+import { DateTimeWidgetProps } from './WidgetEntry.types';
+
+const useStyles = makeStyles<Theme, DateTimeStyleProps>((theme) => ({
+  style: {
+    backgroundColor: ({ backgroundColor }) => (backgroundColor ? getCSSColor(backgroundColor, theme) : null),
+    color: ({ foregroundColor }) => (foregroundColor ? getCSSColor(foregroundColor, theme) : null),
+    fontStyle: ({ italic }) => (italic ? 'italic' : null),
+    fontWeight: ({ bold }) => (bold ? 'bold' : null),
+  },
+  textfield: {
+    marginTop: theme.spacing(0.5),
+    marginBottom: theme.spacing(0.5),
+  },
+  selected: {
+    color: theme.palette.primary.main,
+  },
+}));
+
+export const DateTimeWidget = ({ widget }: DateTimeWidgetProps) => {
+  const props: DateTimeStyleProps = {
+    backgroundColor: widget.style?.backgroundColor ?? null,
+    foregroundColor: widget.style?.foregroundColor ?? null,
+    italic: widget.style?.italic ?? null,
+    bold: widget.style?.bold ?? null,
+  };
+  const classes = useStyles(props);
+
+  const [selected, setSelected] = useState<boolean>(false);
+  const { selection } = useSelection();
+  const ref = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (ref.current && selection.entries.find((entry) => entry.id === widget.id)) {
+      ref.current.focus();
+      setSelected(true);
+    } else {
+      setSelected(false);
+    }
+  }, [selection, widget]);
+
+  const type = widget.isDateOnly ? 'date' : 'datetime-local';
+  const value = widget.isDateOnly ? '2024-06-25' : '2024-06-25T10:30:00';
```

Comment:
What do you mean by real value? They are real value. Possibly, I could use the values of Date.now() if you prefer ?

Reply:
I agree with you for the case of the button label as much as I consider useless for the value field because the specifier will never set a static value. So prefer set a dummy value accordingly to the type of datetime widget, so that something is displayed in the widget rather than an empty value

Reply:
Ok